app/client/ui/userTrustsCustomWidget.ts

```diff
@@ -0,0 +1,71 @@
+import {Computed, dom, makeTestId, Observable, styled} from 'grainjs';
+import {makeT} from 'app/client/lib/localization';
+import {saveModal} from 'app/client/ui2018/modals';
+import {alert} from 'app/client/ui2018/alerts';
+import {labeledSquareCheckbox} from 'app/client/ui2018/checkbox';
+import {inlineMarkdown} from 'app/client/lib/markdown';
+
+const testId = makeTestId('test-custom-widget-warning-modal-');
+
+/**
+ * Show a modal to the user asking him to confirm he trusts the custom widget he's about to install.
+ *
+ * @returns Promise<boolean> Promise that resolves to true if the user confirms he trusts the widget, false otherwise.
+ */
+export function userTrustsCustomWidget() {
+  const t = makeT('userTrustsCustomWidget');
+  const modalT = makeT('modals');
```

Comment:
I added a `Confirm`, generic translation tied to modals, that I use in this file.

I noticed that the `yarn generate:translation` script didn't catch the new translation by doing this.

Should I handle this in another way? I ended adding the translation key manually in the `en.client.json`.

Reply:
You should add the `t('Confirm)` to `modals.ts` and import it here.
The first t should be at the top of the file, the pattern is that we have translatable section per file, and to share things we just export them as normal JS variables.

As an example you can search for `LIMITED_COLUMN_OPTIONS` variable exported that way.

Reply:
Thanks!